src/components/SearchBar/searchbar.js

```diff
@@ -0,0 +1,312 @@
+            </form>
+            <div class="selected-items">
+            </div>
+        </div>
+        `
+  },
+
+  searchFunction: () => {
+    search_div = document.getElementsByClassName('selected-items')
+    const search = document.getElementById('search_btn')
+
+    inp = document.getElementById('search_bar')
+    var currentFocus
+    /* execute a function when someone writes in the text field: */
+    inp.addEventListener('input', async function (e) {
+      var a
+      val = this.value
+
+      if (val.length === 2) {
+        obj = {
+          zipCodes: [],
+          cities: [],
+          streets: [],
+          typos: [],
+          styles: [],
+          intervenants: []
+        }
+        resp = await Api.getAutocomplete('fr', val)
+        SearchBar.addItemsToObj(resp.zipCodes, obj.zipCodes)
+        SearchBar.addItemsToObj(resp.cities, obj.cities)
+        SearchBar.addItemsToObj(resp.streets, obj.streets)
+        SearchBar.addItemsToObj(resp.typos, obj.typos)
+        SearchBar.addItemsToObj(resp.styles, obj.styles)
+        SearchBar.addItemsToObj(resp.intervenants, obj.intervenants)
+      }
+
+      /* close any already open lists of autocompleted values */
+      SearchBar.closeAllLists(inp)
+      if (!val) { return false }
+      currentFocus = -1
+      /* create a DIV element that will contain the items (values): */
+      a = document.createElement('DIV')
+      a.setAttribute('id', this.id + 'autocomplete-list')
+      a.setAttribute('class', 'autocomplete-items')
+      /* append the DIV element as a child of the autocomplete container: */
+      this.parentNode.appendChild(a)
+      /* for each item in the array... */
+      SearchBar.addItemsToList(a, obj.zipCodes, 'Zip code')
+      SearchBar.addItemsToList(a, obj.cities, 'City')
+      SearchBar.addItemsToList(a, obj.streets, 'Street')
+      SearchBar.addItemsToList(a, obj.intervenants, 'Architect', architectIcon, 'search--architect', 'tag--architect')
+      SearchBar.addItemsToList(a, obj.styles, 'Style', styleIcon, 'search--style', 'tag--style')
+      SearchBar.addItemsToList(a, obj.typos, 'Type', typeIcon, 'search--type', 'tag--type')
+    })
+    /* execute a function presses a key on the keyboard: */
+    inp.addEventListener('keydown', function (e) {
+      var x = document.getElementById(this.id + 'autocomplete-list')
+      if (x) x = x.getElementsByTagName('div')
+      if (e.keyCode === 40) {
+        /* If the arrow DOWN key is pressed,
+         increase the currentFocus variable: */
+        currentFocus++
+        /* and and make the current item more visible: */
+        SearchBar.addActive(x, currentFocus)
+      } else if (e.keyCode === 38) { // up
+        /* If the arrow UP key is pressed,
+        decrease the currentFocus variable: */
+        currentFocus--
+        /* and and make the current item more visible: */
+        SearchBar.addActive(x)
+      } else if (e.keyCode === 13) {
```

Comment:
`keyCode` is deprecated (see https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode).